[clang-tidy] Exclude cppcoreguidelines-avoid-magic-numbers

malfet opened the pull request:
Add cppcoreguidelines-avoid-magic-numbers exclusion to clang-tidy
Remove existing nolint warnings using following script:
```
for file in `git ls-files`; do gsed '/\/\/\ NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)/d' -i  $file; done
```

malfet committed:
Exclude cppcoreguidelines-avoid-magic-numbers

janeyx99 commented on .clang-tidy:
```diff
@@ -8,6 +8,7 @@ bugprone-*,
 -bugprone-lambda-function-name,
 -bugprone-reserved-identifier,
 cppcoreguidelines-*,
+-cppcoreguidelines-avoid-magic-numbers,
```
Like remove them from this exclusion list and fix the code that would break due to these

malfet replied:
Some of those yes, but not all. I.e. we are excluding `avoid-magic-numbers` as there are too many false positives

malfet committed:
Delete  NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)

Generated using following shell file:
```
for file in `git ls-files | grep -v \.py`; do gsed '/^ *\/\/ NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)/d' -i  $file; done
```

malfet commented:
Updated the script to delete only NOLINT macros on newline
